boot/bootutil/src/encrypted.c

```diff
@@ -536,7 +536,7 @@ boot_enc_decrypt(const uint8_t *buf, uint8_t *enckey)
      */
     bootutil_ecdh_p256_init(&ecdh_p256);
 
-    rc = bootutil_ecdh_p256_shared_secret(&ecdh_p256, &buf[EC_PUBK_INDEX], private_key, shared);
+    rc = bootutil_ecdh_p256_shared_secret(&ecdh_p256, buf, private_key, shared);
```

Comment:
`&buf[EC_PUBK_INDEX]` != `buf`, can this work? I don't understand why you are trying to change the structure of the data in the buffer.

Reply:
there is a call to mbedtls_ecp_point_read_binary() inside _shared_secret()

It does a PK check for [0]-element to be 0x04, which seems to be different behavior from tinycrypt

I can suggest to make conditional-compile call(s) here - mbedTLS will take whole buff, tinycrypt from 1st element

Reply:
I see, interesting that the library function would check for uncompressed at this place, but then what you should do is to leave the code as it was and instead set `EC_PUBK_INDEX` for MbedTLS to `0` (and leave it as `1` for Tinycrypt).

Reply:
sounds reasonable

Reply:
how do you fee about following approach:

- make bootutil_ecdh_p256_shared_secret() accept beginning of buffer, so complete raw TLV will be passed in
- move if (buf[0] != 0x04) under tinycrypt-specific implementation of bootutil_ecdh_p256_shared_secret();
- mbedTLS-specific implementation will not have this, because internal functions did that

Reply:
same approach for bootutil_ecdh_x25519_shared_secret()

and define different EC_PUBK_INDEX  in ecdh_x25519.h, ecdh_ec256.h respectively, so moved out of encrypted.c